app/services/course_search_service.rb

```diff
@@ -34,36 +34,37 @@ def call
     # The 'where' scope will remove duplicates
     # An outer query is required in the event the provider name is present.
     # This prevents 'PG::InvalidColumnReference: ERROR: for SELECT DISTINCT, ORDER BY expressions must appear in select list'
-    outer_scope = Course.includes(
+    scope = scope.includes(
       :enrichments,
       subjects: [:financial_incentive],
       site_statuses: [:site],
       provider: %i[recruitment_cycle ucas_preferences],
-    ).where(id: scope.select(:id))
```

Comment:
Thanks @defong, I think you are right. I thought I had this working by adding a `DISTINCT(course.id)` to the select list but it's not solving the problem. The `PG::InvalidColumnReference: ERROR: for SELECT DISTINCT, ORDER BY expressions must appear in select list` errors seem to be triggered by this scope: https://github.com/DFE-Digital/teacher-training-api/blob/master/app/models/provider.rb#L97-L103. To be honest this has me pretty confused, I guess it's there to make sure that courses that match on the their own provider appear before those that match the accrediting provider in the results?

I think I can solve the issue by adding the case expression to the select list of the top-level query and then changing the `Course.accredited_body_order` scope to use that rather than `Provider.by_provider_name` (see https://github.com/DFE-Digital/teacher-training-api/pull/2188/commits/f1f60b47986d0c7ab293dc6e6527f106da7aded7). But now I'm still concerned about whether there are other unseen issues that this or the original 'optimisations' have triggered.

Reply:
> To be honest this has me pretty confused, I guess it's there to make sure that courses that match on the their own provider appear before those that match the accrediting provider in the results?

I think so, @TobyRet can you confirm, I sure some it was a an actual business requirement.

Reply:
@defong @TobyRet we are thinking that the best thing to do at this stage is to separate out the two optimisations in this PR so that we can potentially ship the more valuable and less disruptive one (eliminating the double join) before the second one (merging the inner/outer query into one)...

Reply:
Putting this back to draft status for now, in favour of https://github.com/DFE-Digital/teacher-training-api/pull/2197, because it needs more testing